src/main/java/org/opensearch/jobscheduler/model/JobDetails.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject();
+        if (jobIndex != null) {
+            xContentBuilder.field(JOB_INDEX, jobIndex);
+        }
+        if (jobType != null) {
+            xContentBuilder.field(JOB_TYPE, jobType);
+        }
+        if (jobParserAction != null) {
+            xContentBuilder.field(JOB_PARSER_ACTION, jobParserAction);
+        }
+        if (jobRunnerAction != null) {
+            xContentBuilder.field(JOB_RUNNER_ACTION, jobRunnerAction);
+        }
+        return xContentBuilder.endObject();
+    }
+
+    public static JobDetails parse(XContentParser parser) throws IOException {
+        String jobIndex = null;
+        String jobType = null;
+        String jobParserAction = null;
+        String jobRunnerAction = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+            switch (fieldName) {
+                case JOB_INDEX:
+                    jobIndex = parser.text();
+                    break;
+                case JOB_TYPE:
+                    jobType = parser.text();
+                    break;
+                case JOB_PARSER_ACTION:
+                    jobParserAction = parser.text();
+                    break;
+                case JOB_RUNNER_ACTION:
+                    jobRunnerAction = parser.text();
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+
+        return new JobDetails(jobIndex, jobType, jobParserAction, jobRunnerAction);
+    }
+
+    public String getJobIndex() {
+        return jobIndex;
+    }
+
+    public void setJobIndex(String jobIndex) {
+        this.jobIndex = jobIndex;
+    }
+
+    public String getJobType() {
+        return jobType;
+    }
+
+    public void setJobType(String jobType) {
+        this.jobType = jobType;
+    }
+
+    public String getJobParserAction() {
+        return jobParserAction;
+    }
+
+    public void setJobParserAction(String jobParserAction) {
+        this.jobParserAction = jobParserAction;
+    }
+
+    public String getJobRunnerAction() {
+        return jobRunnerAction;
+    }
+
+    public void setJobRunnerAction(String jobRunnerAction) {
+        this.jobRunnerAction = jobRunnerAction;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        JobDetails that = (JobDetails) o;
+        return Objects.equals(jobIndex, that.jobIndex)
+            && Objects.equals(jobType, that.jobType)
+            && jobParserAction.equals(that.jobParserAction)
```

Comment:
Why are these last two a different format than the previous two (`Objects.equals()`)?   Also it looks like they can be `null` so this would throw an NPE, so you would want to use the `Objects` style.

Reply:
ACK